asmjit no longer compilers with within CLR project.

For some reason when the /clr:netcore is used and including the latest version of asmjit in the project it causes compiler errors.

The error is following:
```
3>F:\C++\Dotx64Dbg\testing\_deps\asmjit-src\src\asmjit\x86\../x86/x86operand.h(55,1): error C2088: '|': illegal for struct (compiling source file F:\C++\Dotx64Dbg\src\Bindings\Instruction.Encoder.cpp)
3>F:\C++\Dotx64Dbg\testing\_deps\asmjit-src\src\asmjit\x86\../x86/x86operand.h(55,1): error C2056: illegal expression (compiling source file F:\C++\Dotx64Dbg\src\Bindings\Instruction.Encoder.cpp)
3>F:\C++\Dotx64Dbg\testing\_deps\asmjit-src\src\asmjit\x86\../x86/x86operand.h(56,1): error C2088: '|': illegal for struct (compiling source file F:\C++\Dotx64Dbg\src\Bindings\Instruction.Encoder.cpp)
3>F:\C++\Dotx64Dbg\testing\_deps\asmjit-src\src\asmjit\x86\../x86/x86operand.h(56,1): error C2056: illegal expression (compiling source file F:\C++\Dotx64Dbg\src\Bindings\Instruction.Encoder.cpp)
```
It appears that for some reason when compiled with CLR the `operator|` for OperandSignature cannot be used in the enum.

As a work-around I patched in the way it was done from an older version such as this:
```cpp
//! \cond INTERNAL
//! Adds a template specialization for `REG_TYPE` into the local `RegTraits`.
#define ASMJIT_DEFINE_REG_TRAITS(REG, REG_TYPE, GROUP, SIZE, COUNT, TYPE_ID) \
template<>                                                                   \
struct RegTraits<REG_TYPE> {                                                 \
  typedef REG RegT;                                                          \
                                                                             \
  static constexpr uint32_t kValid = 1;                                      \
  static constexpr uint32_t kCount = COUNT;                                  \
                                                                             \
  static constexpr RegType kType = REG_TYPE;                                 \
  static constexpr RegGroup kGroup = GROUP;                                  \
  static constexpr uint32_t kSize = SIZE;                                    \
                                                                             \
  static constexpr uint32_t kSignature =                                     \
    (OperandSignature::fromOpType(OperandType::kReg) |                       \
     OperandSignature::fromRegType(kType)            |                       \
     OperandSignature::fromRegGroup(kGroup)          |                       \
     OperandSignature::fromSize(kSize)).bits();                              \
                                                                             \
  static constexpr TypeId kTypeId = TYPE_ID;                                 \
}
```

I would propose to do it the old way having static constexpr, the enum has little benefit here and unfortunately breaks my CLR project.

--- a/src/asmjit/core/operand.h
+++ b/src/asmjit/core/operand.h
@@ -1082,19 +1082,19 @@ template<>                                                                   \
 struct RegTraits<REG_TYPE> {                                                 \
   typedef REG RegT;                                                          \
                                                                              \
-  enum : uint32_t {                                                          \
-    kValid = uint32_t(true),                                                 \
-    kCount = uint32_t(COUNT),                                                \
-    kType = uint32_t(REG_TYPE),                                              \
-    kGroup = uint32_t(GROUP),                                                \
-    kSize = uint32_t(SIZE),                                                  \
-    kTypeId = uint32_t(TYPE_ID),                                             \
+  static constexpr uint32_t kValid = 1;                                      \
+  static constexpr uint32_t kCount = COUNT;                                  \
+  static constexpr RegType kType = REG_TYPE;                                 \
+  static constexpr RegGroup kGroup = GROUP;                                  \
+  static constexpr uint32_t kSize = SIZE;                                    \
+  static constexpr TypeId kTypeId = TYPE_ID;                                 \
                                                                              \
-    kSignature = (OperandSignature::fromOpType(OperandType::kReg) |          \
-                  OperandSignature::fromRegType(REG_TYPE)         |          \
-                  OperandSignature::fromRegGroup(GROUP)           |          \
-                  OperandSignature::fromSize(kSize)).bits(),                 \
-  };                                                                         \
+  static constexpr uint32_t kSignature =                                     \
+    (OperandSignature::fromOpType(OperandType::kReg) |                       \
+     OperandSignature::fromRegType(kType)            |                       \
+     OperandSignature::fromRegGroup(kGroup)          |                       \
+     OperandSignature::fromSize(kSize)).bits();                              \
+                                                                             \
 }
 
 //! Adds constructors and member functions to a class that implements abstract register. Abstract register is register
@@ -1135,12 +1135,10 @@ public:                                                                      \
 //! signature.
 #define ASMJIT_DEFINE_FINAL_REG(REG, BASE, TRAITS)                           \
 public:                                                                      \
-  enum : uint32_t {                                                          \
-    kThisType  = TRAITS::kType,                                              \
-    kThisGroup = TRAITS::kGroup,                                             \
-    kThisSize  = TRAITS::kSize,                                              \
-    kSignature = TRAITS::kSignature                                          \
-  };                                                                         \
+  static constexpr RegType kThisType = TRAITS::kType;                        \
+  static constexpr RegGroup kThisGroup = TRAITS::kGroup;                     \
+  static constexpr uint32_t kThisSize  = TRAITS::kSize;                      \
+  static constexpr uint32_t kSignature = TRAITS::kSignature;                 \
                                                                              \
   ASMJIT_DEFINE_ABSTRACT_REG(REG, BASE)                                      \
                                                                              \

